DANS-KNAW/easy-deposit-properties
# EASY-2336 delete datasets with a list of ids

Fixes EASY-2336 delete datasets with a list of ids

#### When applied it will
* 
* 
* 

#### By submitting this pull request I confirm that
* [x] all files contain licenses (`mvn license:format`)
* [x] the project compiles (`mvn clean install`)
* [x] no more new TODO lines
* [x] all unit tests pass (`mvn test`)
  travis succeeds, as well as `mvn clean test`, but in intellij the server tests fail with 

        Caused by: java.lang.ClassNotFoundException: javax.xml.bind.DatatypeConverter
	  at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
	  at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	  ... 29 more
      "[Gateway timeout]" 
* [x] have new examples for unit tests with multiple _existing_ and _non-existing_ datasets
* [x] the changes have been tested on `deasy`

#### Where should the reviewer @DANS-KNAW/easy start?

#### How should this be manually tested?

* start with a clean [base box](http://nexus.dans.knaw.nl/repository/vagrant/deasy-2019-10-22.box)
* checkout dtap at merge of EASY-2379
* check database rights: `sudo -u postgres psql easy_deposit_properties -c "\z"`
* deploy
* check database rights again, each table should now have delete rights:
  `easy_deposit_properties=ar`**d**`/postgres`
* copy-paste-execute scripts from `src/test/resources/graphql-examples/` on http://deasy.dans.knaw.nl:20200/graphiql
  * `delete/notExisting.graphql` should return no depositId
  * `register/plain.graphql`
  * `delete/notExisting.graphql` should return one depositId
  * `delete/notExisting.graphql` should return no depositId
  * `addDeposit/addNewDeposit.graphql`
  * `delete/notExisting.graphql` should return one depositId
  * `delete/notExisting.graphql` should return no depositId


#### Related pull requests on github
repo                       | PR
-------------------------- | -----------------
easy-                      | [PR#](PRlink)
